be/src/storage/lake/general_tablet_writer.cpp

```diff
@@ -239,7 +262,14 @@ StatusOr<std::unique_ptr<SegmentWriter>> VerticalGeneralTabletWriter::create_seg
 
 Status VerticalGeneralTabletWriter::flush_columns(std::unique_ptr<SegmentWriter>* segment_writer) {
     uint64_t index_size = 0;
-    RETURN_IF_ERROR((*segment_writer)->finalize_columns(&index_size));
+    if (_segment_writer_finalize_token != nullptr) {
+        return _segment_writer_finalize_token->submit_func([segment_writer, index_size]() mutable {
+            RETURN_IF_ERROR((*segment_writer)->finalize_columns(&index_size));
```

Comment:
Share `unique_ptr` among differnt threads is an anti-pattern, should change the type from `std::unique_ptr<SegmentWriter>` to `std::shared_ptr<SegmentWriter>`

Reply:
> Share `unique_ptr` among differnt threads is an anti-pattern, should change the type from `std::unique_ptr<SegmentWriter>` to `std::shared_ptr<SegmentWriter>`

`segment_writer`'s type is `std::unique_ptr<SegmentWriter>*`, here the raw pointer was passed directly the task function. The lifecycle of original `std::unique_ptr` was not affected by the whole `flush_columns` function.

Reply:
It may works, but it's not the recommended way. IMHO, shared_ptr is much more applicable here than unique_ptr, and changing the type of `segment_writer` to `std::shared_ptr<SegmentWriter>` shouldn't take much work.

Reply:
Compared with `std::shared_ptr`, `std::unique_ptr` make it harder for programers to reason about the lifecycle management of `segment_writer`.